python/pyarrow/tests/test_compute.py

```diff
@@ -3721,3 +3724,34 @@ def test_pairwise_diff():
     with pytest.raises(pa.ArrowInvalid,
                        match="overflow"):
         pa.compute.pairwise_diff_checked(arr, period=-1)
+
+
+hash_types = st.deferred(
```

Comment:
@kszucs The kernel tests should run on the C++ side, no on the Python side. It does not make sense to add hypothesis tests for this IMHO.

(also, IME nobody knows how to run/debug the hypothesis tests...)

Reply:
I know, hypothesis was quicker to go with, hypothesis is really handy to discover the failing types actually.
I'm currently adding the test cases on the cpp side.

Reply:
Added C++ tests.